File: datasets/Python/Copilot/DataStructures/medium5.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None

def insert_manual():
    val = input("Enter node value (or press Enter for null): ")
    if val == "":
        return None
    node = Node(int(val))
    print(f"Left child of {node.data}:")
    node.left = insert_manual()
    print(f"Right child of {node.data}:")
    node.right = insert_manual()
    return node

def is_bst(node, min_val=float('-inf'), max_val=float('inf')):
    if node is None:
        return True
    if not (min_val < node.data < max_val):
        return False
    return is_bst(node.left, min_val, node.data) and is_bst(node.right, node.data, max_val)

root = insert_manual()
print("Yes" if is_bst(root) else "No")

File: datasets/C++/Copilot/DataStructures/medium5.cpp
#include <iostream>
#include <limits>
#include <climits>
using namespace std;

struct Node {
    int data;
    Node* left;
    Node* right;
    Node(int val) : data(val), left(NULL), right(NULL) {}
};

Node* insertManual(const string& prompt) {
    cout << prompt;
    string input;
    getline(cin, input);
    if (input.empty()) return NULL;
    int val = stoi(input);
    Node* node = new Node(val);
    node->left = insertManual("Left of " + to_string(val) + ": ");
    node->right = insertManual("Right of " + to_string(val) + ": ");
    return node;
}

bool isBST(Node* node, int minVal, int maxVal) {
    if (!node) return true;
    if (node->data <= minVal || node->data >= maxVal) return false;
    return isBST(node->left, minVal, node->data) && isBST(node->right, node->data, maxVal);
}

int main() {
    cout << "Press Enter for null value.\n";
    Node* root = insertManual("Root: ");
    cout << (isBST(root, INT_MIN, INT_MAX) ? "Yes" : "No") << endl;
    return 0;
}
